extensions/oplogPopulator/OplogPopulator.js

```diff
@@ -261,7 +261,11 @@ class OplogPopulator {
             // initialize mongo client
             await this._setupMongoClient();
 
-            if (this._isPipelineImmutable()) {
+            if (this._config.numberOfConnectors === 0) {
+                // If the number of connector is set to 0, then we
+                // use a single connector to listen to the whole DB.
+                this._maximumBucketsPerConnector = Infinity;
```

Comment:
i wonder if it is safer to use this `Infinity` value (vs using `0` as in the config)

semantically, it is more correct indeed ; however since it requires custom handling anyway, maybe having 0 reduces the risk of "missing something" or behaving unexpectedly (e.g. with 0, we will sure it will just fail if we try to go through the regular allocation process, etc...)